Consolidate shared

joverlee521 opened the pull request:
### Description of proposed changes

Based on @genehack's suggestion in our priorities meeting today, this PR consolidates https://github.com/nextstrain/shared with this repo. 

Since I have to do path shuffling in order for pathogen workflows to support external analysis directories with `nextstrain run`, I might as well do all the shuffling upfront instead of propagating out `nextstrain/shared` then do another round of shuffling later down the road.

After this is merged, I'll delete `nextstrain/shared` and rename this repo to `shared`. 

### Related issue(s)

Related to https://github.com/nextstrain/pathogen-repo-guide/issues/83

### Checklist

<!-- Make sure checks are successful at the bottom of the PR. -->

- [x] Checks pass
- [x] If adding a script, add an entry for it in the README.

<!-- 🙌 Thank you for contributing to Nextstrain! ✨ -->

victorlin commented:
Isn't the bulk of the work still to be done within individual pathogen repos? I think it'd be something like this per repo:

1. `git subrepo clone https://github.com/nextstrain/shared shared`
2. apply any pathogen-specific edits from `ingest/vendored` to the new scripts in `shared`
3. update script references in `ingest` to use the new scripts
4. remove `ingest/vendored`

and until that is done, we wouldn't be able to `git subrepo pull` within `ingest/vendored`.

joverlee521 commented:
Deleted the previous `nextstrain/shared` repo and renamed this repo to `shared`. Also pushed up https://github.com/nextstrain/shared/commit/c32a7d31cedb58f87285043a5e69dd9a6c7c0605 to update references from `ingest` to `shared`. 

Will work on updating repos to use this new repo layout.
